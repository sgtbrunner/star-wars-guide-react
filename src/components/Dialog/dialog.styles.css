.overlay {
  position: fixed;
  /* z-index: 100; */
  /* left: 0; */
  top: 0;
  width: 100%;
  height: 100%;
  /* overflow: hidden; */
  background-color: rgba(0, 0, 0, 0.8);
}

.dialog-content,
.dialog-load {
  background-color: #fefefe;
  margin: 15vh auto;
  padding: 4% 4% 4% 3%;
  border: 3px solid #888;
  border-radius: 25px;
  width: 700px;
}

.dialog-load {
  text-align: center;
}

@keyframes spin {
  0% {
    transform: rotate(0deg);
  }
  100% {
    transform: rotate(360deg);
  }
}

.dialog-loader {
  border: 2vw solid #f3f3f3;
  border-top: 2vw solid goldenrod;
  border-radius: 50%;
  width: 15vw;
  height: 15vw;
  animation: spin 2s linear infinite;
  display: inline-block;
}

.stats-list {
  display: grid;
  grid-template-columns: 40% 60%;
  grid-column-gap: 5%;
  /* grid-row-gap: 0; */
}

.close {
  /* color: #aaaaaa; */
  float: right;
  font-size: 45px;
  /* font-weight: bold; */
  /* z-index: 1; */
  user-select: none;
  margin: -4vw -3vw 0 0;
}

.close:hover,
.close:focus {
  color: #000;
  /* text-decoration: none; */
  cursor: pointer;
}

.portrait {
  grid-row: 1 / span 8;
  /* width: 100%; */
  /* height: auto; */
  border: 4px solid goldenrod;
  user-select: none;
}

.char-name {
  font-size: 30px;
  font-weight: bolder;
}

.stats {
  font-size: 18px;
}

@media only screen and (max-width: 768px) {
  .dialog-content,
  .dialog-load {
    width: auto;
    margin: 5vh 5vw;
  }

  .dialog-loader {
    border: 5vw solid #f3f3f3;
    border-top: 5vw solid goldenrod;
    width: 40vw;
    height: 40vw;
  }

  .stats-list {
    grid-template-columns: auto;
    padding: 10px;
    justify-items: center;
    grid-row-gap: 10px;
  }

  .char-name {
    margin-bottom: 2vw;
  }

  .portrait {
    grid-row: 2 / span 7;
    max-width: 240px;
    width: 50vw;
    margin-bottom: 2vw;
  }

  .stats {
    justify-self: start;
  }
}
