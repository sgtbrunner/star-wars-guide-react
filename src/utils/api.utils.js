import { BASE_API_URL } from './constants.utils';

const fetchData = async ({ path, url, resource }) => {
  await fetch(url || `${BASE_API_URL}${path}`, {
    method: 'GET',
    headers: { 'Content-Type': 'application/json' },
  })
    .then((response) => response.json())
    .then(async (data) => {
      resource.push(...data.results);
      if (data.next) await fetchData({ url: data.next, resource });
    });
  return resource;
};

export default {
  getData: (path) => fetchData({ path, resource: [] }),
};
