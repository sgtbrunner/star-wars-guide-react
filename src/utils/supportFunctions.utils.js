// ** These functions support the main App and its Components dealing with
// API, its errors and inconsistencies.  

// VARIABLE DECLARATION
	let pages = [];
	const urls = [
		'https://swapi.dev/api/people/',
		'https://swapi.dev/api/people/?page=2',
		'https://swapi.dev/api/people/?page=3',
		'https://swapi.dev/api/people/?page=4',
		'https://swapi.dev/api/people/?page=5',
		'https://swapi.dev/api/people/?page=6',
		'https://swapi.dev/api/people/?page=7',
		'https://swapi.dev/api/people/?page=8',
		'https://swapi.dev/api/people/?page=9',
	]
	
// FUN FUN FUNCTIONS
// Gets all the characters from different pages and concat them in a single array
const getCharacters = async function() {
	try {
		await getData();			
		return concatArray(pages);
	} catch (err) {
	   	console.log('ooooooops', err);		
	}
}

// Fetches all the character data information from API server and store it in pages as it's defined on SWAPI
const getData = async function() {
	try {
		pages = await Promise.all(urls.map(async function(url) {
	        const response = await fetch(url);
	        return await response.json();
	    }));
	} catch (err) {
	   		console.log('ooooooops', err);	
	}
}

// Function for concating information in an array
const concatArray = (array) => {
	let people = [];
	let i;
	for( i=0 ; i<array.length ; i++ ) {
		people = people.concat(array[i].results);
	}
	return people;
}

// ** This auxiliar function adds obj2 props into obj1 and returns obj1
const augment = (obj1, obj2) => {
	let prop;
	for(prop in obj2) {
		if (obj2.hasOwnProperty(prop) && !obj1[prop]) {
			obj1[prop] = obj2[prop];
		}
	}
	return obj1;
}

// ** This function adds ID property to every object without ID inside a given array of objects
const addId = (array) => {
	array.map((item, index) => {
			const newprop = {id: index}
			return item = augment(item, newprop);
	})
}

export const createList = async function() {
	let list = await getCharacters();
	addId(list);
	return list;
}

// ** SWAPI has an issue handling https://swapi.co/api/people/17/ returnig error 404,
// 	which can be bypassed with the following function
export const getImage = (index) => {
		if (index < 16) {
				return `https://starwars-visualguide.com/assets/img/characters/${(index+1)}.jpg`;
			} else {
				return `https://starwars-visualguide.com/assets/img/characters/${(index+2)}.jpg`;
		}
	}

		// Used for fetching "species" and "homeworld" information from characters
export const getStats = async function(url) {
		// SWAPI hasn't defined "species" nor "homeworld" for some characters, therefore it has to be handled accordingly
		if (url.toString()!=='' && url.toString()!==[] && url.toString()!=={}) {
			try {		
		        const response = await fetch(url);
		        return await response.json();
			} catch (err) {
		   		console.log('ooooooops', err);		
			}	
		} else {
			return await {"name": "unknown"};
		}
	}

	// Used for fetching "films" information from characters
export const getFilms = async function(urls) {
		let movies = '';
		let i=0;
		try {
			for (i=0; i<urls.length; i++) {
				let response = await fetch(urls[i]);
				let movie = await response.json();
				if (i < urls.length-1) {
					movies += movie.title + ', ';
				} else {
					movies += movie.title;
				}
			} 
		} catch (err) {
		   		console.log('ooooooops', err);	
			}
		  	return movies;
};