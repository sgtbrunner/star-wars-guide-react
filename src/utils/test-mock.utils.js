const MALE = 'male';

export const shortCharactersList = [
  {
    id: 0,
    name: 'Anakin Skywalker',
    gender: MALE,
    birth_year: '41.9BBY',
    height: 188,
    mass: 84,
    species: [1],
    homeworld: 1,
    films: [4, 5, 6],
  },
  {
    id: 1,
    name: 'Obi-Wan Kenoby',
    birth_year: '57BBY',
    gender: MALE,
    height: 182,
    mass: 77,
    species: [1],
    homeworld: 3,
    films: [1, 2, 3, 4, 5, 6],
  },
  {
    id: 2,
    name: 'Yoda',
    birth_year: '896BBY',
    gender: MALE,
    height: 66,
    mass: 17,
    species: [2],
    homeworld: 4,
    films: [2, 3, 4, 5, 6],
  },
];

export const longCharactersList = [
  ...shortCharactersList,
  {
    id: 3,
    name: 'Darth Vader',
    birth_year: '41.9BBY',
    gender: MALE,
    height: 202,
    mass: 136,
    species: [1],
    homeworld: 1,
    films: [1, 2, 3, 6],
  },
  {
    id: 4,
    name: 'Luke Skywalker',
    birth_year: '19BBY',
    gender: MALE,
    height: 172,
    mass: 77,
    species: [0],
    homeworld: 1,
    films: [1, 2, 3, 6],
  },
  {
    id: 5,
    name: 'Palpatine',
    birth_year: '82BBY',
    gender: MALE,
    height: 170,
    mass: 75,
    species: [1],
    homeworld: 1,
    films: [2, 3, 4, 5, 6],
  },
];

export const species = [{ name: 'Human' }, { name: `Yoda's species` }];

export const planets = [
  { name: 'Tatooine' },
  { name: 'Naboo' },
  { name: 'Stewjon' },
  { name: 'unknown' },
];

export const films = [
  { title: 'A New Hope' },
  { title: 'The Empire Strikes Back' },
  { title: 'Return of the Jedi' },
  { title: 'The Phantom Menace' },
  { title: 'Attack of the Clones' },
  { title: 'Revenge of the Sith' },
];
