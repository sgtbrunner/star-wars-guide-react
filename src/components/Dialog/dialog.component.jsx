import React, { useEffect } from 'react';
import PropTypes from 'prop-types';

import { getImageUrl, capitalizeFirstLetter } from '../../utils/functions.utils';
import './dialog.styles.css';

const deserializeStatsList = (list, type) =>
  list.map((item) => type[item - 1]?.name || type[item - 1]?.title).join(', ');

const Dialog = ({
  characterId,
  characters,
  films,
  planets,
  species,
  isDialogDataLoaded,
  closeDialog,
}) => {
  const character = characters[characterId];
  const DIALOG_STATS = isDialogDataLoaded && [
    {
      name: 'Species',
      value: deserializeStatsList(character.species, species) || 'Human',
    },
    { name: 'Gender', value: capitalizeFirstLetter(character.gender) },
    { name: 'Birth Year', value: character.birth_year },
    { name: 'Height', value: character.height, unit: 'cm' },
    { name: 'Weight', value: character.mass, unit: 'Kg' },
    { name: 'Homeworld', value: planets[character.homeworld - 1].name },
    { name: 'Films', value: deserializeStatsList(character.films, films) },
  ];

  useEffect(() => {
    const events = ['click', 'keydown'];
    const closeOnEvent = (event) => {
      if (event.keyCode === 27 || event.target.id === 'overlay') closeDialog();
    };

    events.forEach((event) => window.addEventListener(event, closeOnEvent));
    return () => events.forEach((event) => window.removeEventListener(event, closeOnEvent));
  }, [closeDialog]);

  return (
    <div className="overlay" id="overlay" role="dialog">
      {!isDialogDataLoaded ? (
        <div className="dialog-load" role="alert">
          <div className="dialog-loader" />
        </div>
      ) : (
        <div className="dialog-content">
          <span className="close">
            <button type="button" className="bn pointer bg-transparent" onClick={closeDialog}>
              &times;
            </button>
          </span>
          <div className="stats-list">
            <img src={getImageUrl(character.id)} className="portrait" alt={character.id} />
            <div className="char-name">{character.name}</div>
            {DIALOG_STATS.map((stat) => (
              <div className="stats" key={stat.name} role="listitem">
                <u>{stat.name}</u>
                {`: ${stat.value} ${stat.unit && stat.value !== 'unknown' ? stat.unit : ''}`}
              </div>
            ))}
          </div>
        </div>
      )}
    </div>
  );
};

Dialog.propTypes = {
  characterId: PropTypes.number.isRequired,
  characters: PropTypes.array.isRequired,
  films: PropTypes.array.isRequired,
  planets: PropTypes.array.isRequired,
  species: PropTypes.array.isRequired,
  isDialogDataLoaded: PropTypes.bool.isRequired,
  closeDialog: PropTypes.func.isRequired,
};

export default Dialog;
