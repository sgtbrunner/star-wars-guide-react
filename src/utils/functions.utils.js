import { BASE_IMAGE_URL } from './constants.utils';

export const getIdParamFromUrl = (url) => parseInt(url?.replace(/\D/g, ''), 10);

const mapArrayPropertyToId = (item) => getIdParamFromUrl(item);

export const capitalizeFirstLetter = (string) => string[0].toUpperCase() + string.slice(1);

export const getImageUrl = (index) =>
  typeof index === 'number' || (typeof index === 'string' && Number(index))
    ? `${BASE_IMAGE_URL}${Number(index) < 16 ? Number(index) + 1 : Number(index) + 2}.jpg`
    : null;

export const updateCharactersProperties = (data) =>
  data.map((character, index) => ({
    id: index,
    ...character,
    homeworld: getIdParamFromUrl(character.homeworld),
    films: character.films.map(mapArrayPropertyToId),
    species: character.species.map(mapArrayPropertyToId),
  }));
