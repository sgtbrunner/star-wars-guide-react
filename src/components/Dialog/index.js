import { connect } from 'react-redux';

import { closeDialog } from '../../redux/dialog/dialog.actions';
import Dialog from './dialog.component';

const isDialogDataLoaded = (state) => {
  const { films, planets, species } = state;
  return [films.data.length, planets.data.length, species.data.length].every(
    (itemLength) => itemLength
  );
};

const mapStateToProps = (state) => ({
  characterId: state.dialog.characterId,
  characters: state.characters.data,
  films: state.films.data,
  planets: state.planets.data,
  species: state.species.data,
  isDialogDataLoaded: isDialogDataLoaded(state),
});

const mapDispatchToProps = (dispatch) => ({
  closeDialog: () => closeDialog(dispatch),
});

export default connect(mapStateToProps, mapDispatchToProps)(Dialog);
