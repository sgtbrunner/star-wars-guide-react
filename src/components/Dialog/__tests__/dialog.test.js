import React from 'react';
import { render, screen } from '@testing-library/react';

import Dialog from '../dialog.component';
import { longCharactersList, films, planets, species } from '../../../utils/test-mock.utils';

const closeDialog = jest.fn();

const initialProps = {
  characterId: 0,
  characters: [],
  films: [],
  planets: [],
  species: [],
  isDialogDataLoaded: false,
};

const loadedCharacterProps = {
  characters: longCharactersList,
  films,
  planets,
  species,
  isDialogDataLoaded: true,
};

const loadedYodaCharacterProps = {
  ...loadedCharacterProps,
  characterId: 2,
};

const loadedVaderCharacterProps = {
  ...loadedCharacterProps,
  characterId: 3,
};

const mountDialogComponent = (newProps = initialProps) =>
  render(<Dialog closeDialog={closeDialog} {...newProps} />);

describe('Dialog component', () => {
  it('should match snapshot when data is loading', () => {
    const wrapper = mountDialogComponent();

    expect(wrapper).toMatchSnapshot();
  });

  it('should match snapshot when data is loaded', () => {
    const wrapper = mountDialogComponent(loadedYodaCharacterProps);

    expect(wrapper).toMatchSnapshot();
  });

  it('should render dialog', () => {
    mountDialogComponent();

    const dialog = screen.getByRole('dialog');
    expect(dialog).toBeInTheDocument();
  });

  it('should render loader alert within dialog when data is loading', () => {
    mountDialogComponent();

    const loader = screen.getByRole('alert');
    expect(loader).toBeInTheDocument();
  });

  it('should render a close button within dialog when data is loaded', () => {
    mountDialogComponent(loadedYodaCharacterProps);

    const closeButton = screen.getByRole('button', { name: 'Ã—' });
    expect(closeButton).toBeInTheDocument();
  });

  it(`should render a portrait image and a matched name for Yoda character within 
    dialog when Yoda data is loaded`, () => {
    mountDialogComponent(loadedYodaCharacterProps);

    const portrait = screen.getByRole('img');
    const name = screen.getByText('Yoda');

    expect(portrait).toBeInTheDocument();
    expect(name).toBeInTheDocument();
  });

  it(`should render a portrait image and a matched name for Darth Vader character within 
    dialog when Darth Vader data is loaded`, () => {
    mountDialogComponent(loadedVaderCharacterProps);

    const portrait = screen.getByRole('img');
    const name = screen.getByText('Darth Vader');

    expect(portrait).toBeInTheDocument();
    expect(name).toBeInTheDocument();
  });

  it('should render 7 (seven) stats within dialog when any character data is loaded', () => {
    mountDialogComponent(loadedYodaCharacterProps);

    const stats = screen.getAllByRole('listitem');
    expect(stats).toHaveLength(7);
  });
});
