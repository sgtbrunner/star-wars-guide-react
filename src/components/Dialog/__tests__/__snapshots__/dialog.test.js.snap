// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dialog component should match snapshot when data is loaded 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="overlay"
        id="overlay"
        role="dialog"
      >
        <div
          class="dialog-content"
        >
          <span
            class="close"
          >
            <button
              class="bn pointer bg-transparent"
              type="button"
            >
              ×
            </button>
          </span>
          <div
            class="stats-list"
          >
            <img
              alt="2"
              class="portrait"
              src="https://starwars-visualguide.com/assets/img/characters/3.jpg"
            />
            <div
              class="char-name"
            >
              Yoda
            </div>
            <div
              class="stats"
              role="listitem"
            >
              <u>
                Species
              </u>
              : Yoda's species 
            </div>
            <div
              class="stats"
              role="listitem"
            >
              <u>
                Gender
              </u>
              : Male 
            </div>
            <div
              class="stats"
              role="listitem"
            >
              <u>
                Birth Year
              </u>
              : 896BBY 
            </div>
            <div
              class="stats"
              role="listitem"
            >
              <u>
                Height
              </u>
              : 66 cm
            </div>
            <div
              class="stats"
              role="listitem"
            >
              <u>
                Weight
              </u>
              : 17 Kg
            </div>
            <div
              class="stats"
              role="listitem"
            >
              <u>
                Homeworld
              </u>
              : unknown 
            </div>
            <div
              class="stats"
              role="listitem"
            >
              <u>
                Films
              </u>
              : The Empire Strikes Back, Return of the Jedi, The Phantom Menace, Attack of the Clones, Revenge of the Sith 
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="overlay"
      id="overlay"
      role="dialog"
    >
      <div
        class="dialog-content"
      >
        <span
          class="close"
        >
          <button
            class="bn pointer bg-transparent"
            type="button"
          >
            ×
          </button>
        </span>
        <div
          class="stats-list"
        >
          <img
            alt="2"
            class="portrait"
            src="https://starwars-visualguide.com/assets/img/characters/3.jpg"
          />
          <div
            class="char-name"
          >
            Yoda
          </div>
          <div
            class="stats"
            role="listitem"
          >
            <u>
              Species
            </u>
            : Yoda's species 
          </div>
          <div
            class="stats"
            role="listitem"
          >
            <u>
              Gender
            </u>
            : Male 
          </div>
          <div
            class="stats"
            role="listitem"
          >
            <u>
              Birth Year
            </u>
            : 896BBY 
          </div>
          <div
            class="stats"
            role="listitem"
          >
            <u>
              Height
            </u>
            : 66 cm
          </div>
          <div
            class="stats"
            role="listitem"
          >
            <u>
              Weight
            </u>
            : 17 Kg
          </div>
          <div
            class="stats"
            role="listitem"
          >
            <u>
              Homeworld
            </u>
            : unknown 
          </div>
          <div
            class="stats"
            role="listitem"
          >
            <u>
              Films
            </u>
            : The Empire Strikes Back, Return of the Jedi, The Phantom Menace, Attack of the Clones, Revenge of the Sith 
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Dialog component should match snapshot when data is loading 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="overlay"
        id="overlay"
        role="dialog"
      >
        <div
          class="dialog-load"
          role="alert"
        >
          <div
            class="dialog-loader"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="overlay"
      id="overlay"
      role="dialog"
    >
      <div
        class="dialog-load"
        role="alert"
      >
        <div
          class="dialog-loader"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
